app/controllers/provider/admin/keys_controller.rb

```diff
@@ -14,9 +14,13 @@ def edit
 
   # user key
   def update
-    @cinstance.user_key = params[:cinstance][:user_key]
-    unless @cinstance.save
-      flash.now[:error] = "Invalid key. Please review and try submitting again."
+    user_key = params[:cinstance][:user_key]
```

Comment:
```suggestion
    if user_key.empty?
      @cinstance.errors.add(:user_key, :blank)
    else
      @cinstance.user_key = user_key
      @cinstance.save
    end
    flash.now[:error] = @cinstance.errors.full_messages.to_sentence if @cinstance.errors.present?
```

The suggestion is not really accurate, because there the final `end` is not included.

But basically the whole point is - rely on standard "invalid" or "blank" messages.